src/components/artifacts/artifact-node-sheet/DetailCards.tsx

```diff
@@ -14,95 +19,131 @@ import {
+	if (artifactVersion.isPending || pipelineRun.isPending)
+		return <Skeleton className="h-[500px] w-full" />;
+	if (artifactVersion.isError || pipelineRun.isError)
+		return <ErrorFallback err={artifactVersion.error! || pipelineRun.error!} />;
 
 	return (
 		<CollapsibleCard initialOpen title="Details">
 			<dl className="grid grid-cols-1 gap-x-[10px] gap-y-2 md:grid-cols-3 md:gap-y-4">
-				{artifactVersion.metadata?.producer_step_run_id && (
-					<KeyValue
-						label="Producer Step"
-						value={
+				<KeyValue
+					label="Pipeline"
+					value={
+						<Link
+							to={routes.pipelines.namespace(
+								encodeURIComponent(pipelineRun.data.body?.pipeline?.name as string)
+							)}
+						>
 							<Tag
-								color="grey"
+								color="purple"
 								className="inline-flex items-center gap-0.5"
 								rounded={false}
 								emphasis="subtle"
 							>
-								{stepData?.body?.status === "running" ? (
-									<Spinner className="mr-1 h-4 w-4 border-[2px]" />
-								) : (
-									<ArtifactIcon
-										artifactType={artifactVersion.body?.type as ArtifactVersionBody["type"]}
-										className="mr-1 h-4 w-4 fill-current"
-									/>
-								)}
-								{stepData ? stepData?.name : <Skeleton className="h-5 w-5" />}
+								<Pipelines className="mr-1 h-4 w-4 fill-theme-text-brand" />
+								{pipelineRun.data.body?.pipeline?.name}
+
+								<div className="rounded-sm bg-primary-50 px-1 py-0.25">
+									{pipelineRun.data.body?.pipeline?.body?.version}
+								</div>
 							</Tag>
+						</Link>
+					}
+				/>
+				{artifactVersion.data.body?.producer_pipeline_run_id && pipelineRun.data.body?.status && (
+					<KeyValue
+						label="Producer Run"
+						value={
+							<Link to={routes.runs.detail(artifactVersion.data.body?.producer_pipeline_run_id)}>
+								<Tag
+									color={getExecutionStatusTagColor(pipelineRun.data.body?.status)}
+									className="inline-flex items-center gap-0.5"
+									rounded={false}
+									emphasis="subtle"
+								>
+									{pipelineRun.data.body?.status === "running" ? (
+										<Spinner className="mr-1 h-4 w-4 border-[2px]" />
+									) : (
+										<Run className={`mr-1 h-4 w-4 fill-current`} />
+									)}
+
+									{artifactVersion.data.body?.producer_pipeline_run_id}
+								</Tag>
+							</Link>
 						}
 					/>
 				)}
-				{artifactVersion.body?.producer_pipeline_run_id && (
+				{artifactVersion.data.body?.artifact.id && (
 					<KeyValue
-						label="Producer Run"
+						label="Producer Step"
 						value={
-							<Tag
-								color={getExecutionStatusTagColor(stepData?.body?.status)}
-								className="inline-flex items-center gap-0.5"
-								rounded={false}
-								emphasis="subtle"
-							>
-								{stepData?.body?.status === "running" ? (
-									<Spinner className="mr-1 h-4 w-4 border-[2px]" />
+							<>
+								{stepDetail.data ? (
+									<Tag
+										color={getExecutionStatusTagColor(pipelineRun.data.body?.status)}
```

Comment:
This needs to be double checked

Reply:
@schustmi If i recall correctly thats the part, where we just link a step, but dont need the pipeline status for it right?

Reply:
Yes, the step run should not be coupled to the pipeline run

Reply:
just make it green always, there is no way the artifact exists without the producer step having finished, at least in the current dag visualizer implementation